### PR TITLE
feat(footer): added locale-button to footer

### PR DESCRIPTION
**Pull request by photodow**
### Related Ticket(s)

- [#1640](https://github.ibm.com/webstandards/digital-design/issues/1640)
- [#1078](https://github.ibm.com/webstandards/digital-design/issues/1078)
- [#1485](https://github.ibm.com/webstandards/digital-design/issues/1485)
- [#1368](https://github.ibm.com/webstandards/digital-design/issues/1368)

### Description

This pull request makes some progress forwards for the local selector and places it inside the footer behind a feature flag. Clearly more work needs to be done, but this has already helped scope down some of the prototype and animation exploration work.

image

### Changelog

**New**

- Feature flagged locale selector button in the footer

### CODE REVIEW

**Comment by jeffchew** [comment]
It looks like the output for storybook is failing:
Screen Shot 2019-09-10 at 10 49 48 AM

**Comment by photodow** [comment]
@jeffchew thanks I'm going to try and get this fixed this morning.

**Comment by wonilsuhibm** [comment]
@photodow  i'm getting a `Minified React error` error when i open the netlify url.

File: packages/react/src/internal/FeatureFlags.js [hunk 1]
```diff
@@ -9,9 +9,22 @@
...
 
+/**
+ * Helper function for rendering feature flag
+ *
+ * @param {string} flag identifies which flag is used
+ * @param {string} jsx pass in the jsx to render conditionally
+ * @private
+ * @returns {object} JSX object
+ */
+export const featureFlag = (flag, jsx) => {
```
Comment (jeffchew) [inline packages/react/src/internal/FeatureFlags.js]: @photodow I dig this!

cc: @annawen1 @kennylam

Comment (annawen1) [reply packages/react/src/internal/FeatureFlags.js]: can we pull this out and into the `utilities` package instead? Would want to use this for my lead space component in patterns

Comment (photodow) [reply packages/react/src/internal/FeatureFlags.js]: done and done

File: packages/react/src/components/Footer/LocaleButton.js [hunk 2]
```diff
@@ -1,7 +1,5 @@
-import {
-  featureFlag,
-  FOOTER_LOCALE_BUTTON,
-} from '../../internal/FeatureFlags.js';
+import { FOOTER_LOCALE_BUTTON } from '../../internal/FeatureFlags.js';
+import { featureFlag } from '@carbon/ibmdotcom-utilities';
```
Comment (jeffchew) [inline packages/react/src/components/Footer/LocaleButton.js]: @photodow can you publish an alpha version of this from your local then reference that in package dependencies? 
cc: @annawen1

File: packages/react/src/components/Footer/README.md [hunk 3]
```diff
@@ -24,6 +24,18 @@ ReactDOM.render(<App />, document.querySelector('#app'));
 > 💡 Don't forget to import the footer styles from
 > [@carbon/ibmdotcom-styles](/packages/styles).
 
+#### Feature Flags
+
+To utilize the following features, set the following variable's to `true` within
+your `.env` file.
```
Comment (jeffchew) [inline packages/react/src/components/Footer/README.md]: > To utilize the following features, set the following variable's to `true` within
your `.env` file.

... or your application build settings.

Comment (photodow) [reply packages/react/src/components/Footer/README.md]: Added, thanks!

File: packages/react/src/components/Footer/LocaleButton.js (hunk 2 above)
Comment (photodow) [reply packages/react/src/components/Footer/LocaleButton.js]: looks like it's done. thanks!

**Comment by kennylam** [comment]
> @kennylam @annawen1 any other thoughts on this PR? Looks like with the env variables in place the switch is working fine.

looks good to me

**Pull request merged**